upgrade/upgrade.go

```diff
@@ -217,69 +220,64 @@ func upgradeLocalNode() (err error) {
 	return nil
 }
 
-func doCRUpgrade(namespace string, lhClient *lhclientset.Clientset, kubeClient *clientset.Clientset) (err error) {
+func doResourceUpgrade(namespace string, lhClient *lhclientset.Clientset, kubeClient *clientset.Clientset) (err error) {
 	defer func() {
 		err = errors.Wrap(err, "upgrade CRD failed")
 	}()
-	if err := v070to080.UpgradeCRs(namespace, lhClient); err != nil {
-		return err
-	}
-	if err := v100to101.UpgradeCRs(namespace, lhClient); err != nil {
-		return err
-	}
-	if err := v102to110.UpgradeCRs(namespace, lhClient); err != nil {
-		return err
-	}
-	if err := v110to111.UpgradeCRs(namespace, lhClient); err != nil {
+
+	lhVersionBeforeUpgrade, err := upgradeutil.GetCurrentLonghornVersion(namespace, lhClient)
+	if err != nil {
 		return err
 	}
-	if err := v110to120.UpgradeCRs(namespace, lhClient, kubeClient); err != nil {
-		return err
+
+	if semver.Compare(lhVersionBeforeUpgrade, "v0.8.0") < 0 {
```

Comment:
What's the result if `lhVersionBeforeUpgrade` is empty?

Reply:
If the 1st input is invalid or empty, the result will be `-1` :
https://go.dev/play/p/Xr2fwc-t4Nv